packages/merge-styles/src/transforms/provideUnits.ts

```diff
@@ -12,11 +12,12 @@ const NON_PIXEL_NUMBER_PROPS = [
 ];
 
 export function provideUnits(rulePairs: (string | number)[], index: number): void {
-  const name = rulePairs[index];
+  const name = rulePairs[index] as string;
   const value = rulePairs[index + 1];
 
   if (typeof value === 'number') {
-    const unit = NON_PIXEL_NUMBER_PROPS.indexOf(name as string) === -1 ? 'px' : '';
+    const convertToPixel = NON_PIXEL_NUMBER_PROPS.indexOf(name as string) === -1 && name.indexOf('--') === -1;
```

Comment:
i think we should create a `isCssVariable(name: string): boolean` helper in merge-styles and use it here. most likely will be handy later as well

Reply:
`--webkit-foo` is not a css variable but would match here as well. `isCssVariableOrVendorPrefixed` ?

Reply:
ah ic. either that or do `const isCssVariableOrVendorPrefixed = name.indexOf('--') === -1` here for how. goal is to make this piece code more understandable

Reply:
Updated code to be more clear, good suggestion.